### PR TITLE
Remove bluebird

### PR DESCRIPTION
**Pull request by nacimgoura**
bluebird is useless with nodejs 8 because promises are used natively.

### CODE REVIEW

File: controllers/user.js [hunk 1]
```diff
@@ -1,9 +1,11 @@
-const bluebird = require('bluebird');
-const crypto = bluebird.promisifyAll(require('crypto'));
+const { promisify } = require('util');
+const crypto = require('crypto');
 const nodemailer = require('nodemailer');
 const passport = require('passport');
 const User = require('../models/User');
 
+const randomBytesAsync = promisify(crypto.randomBytes);
```
Comment (YasharF) [inline controllers/user.js]: Can we have the same compact one line definition as before?  Why or why not?

Comment (nacimgoura) [reply controllers/user.js]: This is not possible according to the [documentation](https://nodejs.org/dist/latest-v8.x/docs/api/util.html#util_util_promisify_original).

**Comment by YasharF** [comment]
If you haven't already. can you do the following functional tests manually in the browser and report here?

- Does the password reset process still work?  Asking, since I think crypto related code for that has been touched.  Can you verify in the DB during the process that there is in fact a randomized token present?
- Can you manually verify in the browser that Instagram, FourSquare, and Steam API examples still work, while using working API keys from the three providers, and using real accounts.  You may need more than one account if there are edge cases (aka like the Steam achievement issue in the other PR).

**Comment by nacimgoura** [comment]
I ran the tests again.
the password reset process generates a value like this `5306492510cc17dc16e7040c54488f90`.

But for steam I had created my account specifically to test the project, so I have no achievement, which allowed me to find bug #810. Since I still don't have any achievements and I don't have an account with achievements, I can't test this particular case.

**Comment by YasharF** [comment]
I retested for Steam, Foursquare, and password reset.  No issues.

I did not test for Instagram, due to issues that I ran into with getting permissions through Instagram developer portal, and going to skip testing it since Hackathon Starter readme doesn't even cover it,  Instagram's current APIs are on the deprecation path, and that the npm package is unmaintained.

Thank you!

**Comment by YasharF** [comment]
Note: in-part to replace https://github.com/sahat/hackathon-starter/pull/814
